parser.temper

```diff
@@ -79,154 +81,154 @@ class Parser {
   }
 
   charClassUnit(): CodePart | Bubble {
-    if (matchChar("\\")) {
+    if (matchChar(char "\\")) {
       finishEscape().as<CodePart>() orelse do {
         error("invalid code part");
         bubble()
       }
     } else {
-      new CodePoints(read())
+      new CodePoints(String.fromCodePoint(read()))
     }
   }
 
   extractCode(codePart: CodePart): Int | Bubble {
-    codePart.as<CodePoints>().value.codePoints.read() orelse do {
+    codePart.as<CodePoints>().value.codeAt(String.start) orelse do {
       error("invalid range edge");
       bubble()
     }
   }
 
   finishEscape(): RegexNode | Bubble {
     var escapeCode = peek();
-    if (matchChar("b")) {
+    if (matchChar(char"b")) {
       WordBoundary
-    } else if (matchChar("d")) {
+    } else if (matchChar(char"d")) {
       Digit
-    } else if (matchChar("n")) {
+    } else if (matchChar(char"n")) {
       new CodePoints("\n")
-    } else if (matchChar("r")) {
+    } else if (matchChar(char"r")) {
       new CodePoints("\r")
-    } else if (matchChar("s")) {
+    } else if (matchChar(char"s")) {
       Space
-    } else if (matchChar("t")) {
+    } else if (matchChar(char"t")) {
       new CodePoints("\t")
     // TODO Need more support in temper core for hex decoding.
-    // } else if (matchChar("u")) {
-    //   if (matchChar("{")) {
+    // } else if (matchChar(char"u")) {
+    //   if (matchChar(char"{")) {
     //     //
     //   }
-    } else if (matchChar("w")) {
+    } else if (matchChar(char"w")) {
       Word
     } else if (
-      matchChar("(") || matchChar(")")
-      || matchChar("[") || matchChar("]")
-      || matchChar("{") || matchChar("}")
-      || matchChar(".") || matchChar("\\") || matchChar("-")
+      matchChar(char"(") || matchChar(char")")
+      || matchChar(char"[") || matchChar(char"]")
+      || matchChar(char"{") || matchChar(char"}")
+      || matchChar(char".") || matchChar(char "\\") || matchChar(char"-")
       || matchRepeat()
     ) {
-      new CodePoints(escapeCode)
+      new CodePoints(String.fromCodePoint(escapeCode))
```

Comment:
Some reorg might be able to avoid intermediate objects like this, but I didn't restructure anything seriously in this pr.